src/dsp_tools/commands/excel2json/json_header.py

```diff
@@ -0,0 +1,249 @@
+    if compliance_problem := _do_formal_compliance(sheets_df_dict):
+        raise InputError(compliance_problem.execute_error_protocol())
+    result = _process_file(sheets_df_dict)
+    if isinstance(result, ExcelFileProblem):
+        raise InputError(result.execute_error_protocol())
+    print("json_header.xlsx file used to construct the json header.")
+    return result
+
+
+def _do_formal_compliance(df_dict: dict[str, pd.DataFrame]) -> ExcelFileProblem | None:
+    expected_sheets = ["prefixes", "project", "description", "keywords"]
+
+    def _check_df(sheet: str) -> str | None:
+        if (df := df_dict.get(sheet)) is None:
+            return sheet
+        if len(df) == 0:
+            return sheet
+        return None
+
+    if empty := [x for x in expected_sheets if _check_df(x)]:
+        return ExcelFileProblem("json_header.xlsx", [EmptySheetsProblem(empty)])
+    return None
+
+
+def _process_file(df_dict: dict[str, pd.DataFrame]) -> ExcelFileProblem | FilledJsonHeader:
+    problems: list[Problem] = []
+    prefix_result = _do_prefixes(df_dict["prefixes"])
+    if isinstance(prefix_result, ExcelSheetProblem):
+        problems.append(prefix_result)
+    project_result = _do_project(df_dict)
+    if not isinstance(project_result, Project):
+        problems.extend(project_result)
+    if problems:
+        return ExcelFileProblem("json_header.xlsx", problems)
+    project = cast(Project, project_result)
+    prefixes = cast(Prefixes, prefix_result)
+    return FilledJsonHeader(project, prefixes)
+
+
+def _do_prefixes(df: pd.DataFrame) -> ExcelSheetProblem | Prefixes:
+    if missing_cols := check_contains_required_columns(df, {"prefixes", "uri"}):
+        return ExcelSheetProblem("prefixes", [missing_cols])
+    if missing_vals := check_required_values_get_position_in_excel(df, ["prefixes", "uri"]):
+        return ExcelSheetProblem("prefixes", [missing_vals])
+    pref = dict(zip(df["prefixes"], df["uri"]))
+    pref = {k.rstrip(":"): v for k, v in pref.items()}
+    return Prefixes(pref)
+
+
+def _do_project(df_dict: dict[str, pd.DataFrame]) -> list[ExcelSheetProblem] | Project:
+    problems = []
+    project_df = df_dict["project"]
+    if project_problem := _check_project_sheet(project_df):
+        problems.append(project_problem)
+    description_result = _do_description(df_dict["description"])
+    if isinstance(description_result, ExcelSheetProblem):
+        problems.append(description_result)
+    keywords_result = _do_keywords(df_dict["keywords"])
+    if isinstance(keywords_result, ExcelSheetProblem):
+        problems.append(keywords_result)
+    all_users = None
+    if (user_df := df_dict.get("users")) is not None:
+        if len(user_df) > 0:
+            user_result = _do_users(user_df)
+            match user_result:
+                case ExcelSheetProblem():
+                    problems.append(user_result)
+                case _:
+                    all_users = user_result
```

Comment:
above, instance checking was done with `isinstance`, here you do pattern matching. Both options are viable, but I would keep it consistent.